toolchain/parser/parser.cpp

```diff
@@ -1100,6 +1150,15 @@ auto Parser::HandleFunctionSignatureFinishState() -> void {
       break;
     }
     case TokenKind::OpenCurlyBrace(): {
+      if (stack_context_ == ParseContext::Interface) {
```

Comment:
You can leave this for now, but I want to note, we may end up wanting to move this to semantics: `default fn` can provide a definition, and so the context becomes much more complex (not just interface or not).